### PR TITLE
Promote all IDL from ASCII to UTF-8

### PR DESCRIPTION
**Pull request by kriskowal**
This change replaces the default encoding for Thrift IDL files to UTF-8 for parity with ThriftRW-Go. As UTF-8 is a strict superset of ASCII, this is a backward-compatible change. This change also introduces an encoding option to the Thrift constructor and Thrift.load function, and the setting applies through transitive includes.

The change has ramifications for the new asynchronous loader, Thrift.load, which requires a file system implementation that implements `readFile(path, encoding, cb)` instead of `readFile(path, cb)`, but was not previously released. The Node.js `fs` module satisfies this contract just as well, but this change would have broken any user that provided a less sophisticated filesystem shim.

Fixes #151

### CODE REVIEW

**Comment by kriskowal** [comment]
cc @KeitaMoromizato for their contributions, for review.

cc @dbousque for the change to the async filesystem API necessary to facilitate alternate encodings.

File: thrift.js [hunk 1]
```diff
@@ -111,6 +111,8 @@ Thrift.prototype._init = function _init(options) {
     this.asts = options.asts || Object.create(null);
     // filename to Thrift instance
     this.memo = options.memo || Object.create(null);
+    // file encoding
+    this.encoding = options.encoding || 'ascii'
```
Comment (dbousque) [inline thrift.js]: This line is duplicated on line 128 with `utf-8` instead of `ascii`.

File: thrift.js [hunk 2]
```diff
@@ -177,13 +180,13 @@ Thrift.prototype._asyncParse = function _asyncParse(filename, allowIncludeAlias,
         return model._asyncParseIncludedModules(filename, allowIncludeAlias, cb);
     }
 
-    model.fs.readFile(filename, function (err, source) {
+    model.fs.readFile(filename, this.encoding, function (err, source) {
```
Comment (dbousque) [inline thrift.js]: Is it normal that the encoding is passed as argument twice here?

**Comment by dbousque** [comment]
If the `encoding` argument to `model.fs.load` is required (looks like it is), I guess we should update the documentation in the `Non filesystem and asynchronous source loading` part.

**Comment by kriskowal** [comment]
Having slept on this change, I think it might make more sense to just treat all IDL as UTF-8 and not make the encoding configurable. That would avoid any possible confusion between Thrift compilers provided they converge on UTF-8, which is the only sensible direction up from ASCII today, if we have a prayer of having inter-language interoperability.

File: thrift.js (hunk 2 above)
Comment (kriskowal) [reply thrift.js]: nope. ty.

File: thrift.js (hunk 1 above)
Comment (kriskowal) [reply thrift.js]: addressed. ty.

File: thrift.js [hunk 3]
```diff
@@ -177,13 +177,13 @@ Thrift.prototype._asyncParse = function _asyncParse(filename, allowIncludeAlias,
         return model._asyncParseIncludedModules(filename, allowIncludeAlias, cb);
     }
 
-    model.fs.readFile(filename, function (err, source) {
+    model.fs.readFile(filename, 'utf-8', function (err, source) {
```
Comment (dbousque) [inline thrift.js]: The `model.fs.readFile` can be a custom function, can we update the documentation in the README to inform users that they need to accept three arguments (`Non filesystem and asynchronous source loading`)?

**Pull request merged**